### PR TITLE
Allows empty `website` field

### LINKED ISSUE #153: Make `website` field optional from users
The [frontend](https://my.realdevsquad.com/signup) says that the `website` is optional but if we send blank in website field, the backend sends that `website` cannot be empty and hence form cannot be submitted

### PR DESCRIPTION
**Pull request by swarajpure**
Fixes #153

### CODE REVIEW

File: middlewares/validators/user.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ const updateUser = async (req, res, next) => {
     site: joi.string().optional(),
     isMember: joi.boolean().optional(),
     tokens: joi.object().optional(),
-    website: joi.string().optional()
+    website: joi.string().optional().allow('')
```
Comment (ankurnarkhede) [inline middlewares/validators/user.js:55]: Why is this change being made?
The website name shouldn't be an empty string, right?

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

File: middlewares/validators/user.js (hunk 1 above)
Comment (swarajpure) [reply middlewares/validators/user.js:55]: Yes. Currently the signup form has `website` as optional and when the user leaves it blank, the backend sends 400 response saying `website cannot be empty` 

Source: https://github.com/sideway/joi/blob/v10.6.0/API.md#string---inherits-from-any
image

Comment (ankurnarkhede) [reply middlewares/validators/user.js:55]: Yeah, but this doesn't mean that we should explicitly allow the empty strings. This is an issue from the UI. The UI shouldn't be sending empty strings for website value.

**Comment by swarajpure** [comment]
Handling on the frontend

**Pull request closed without merging**